lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -252,17 +252,32 @@ spec = do
         reward <- getFromResponse balanceReward <$>
             request @ApiWallet ctx (getWalletEp w) Default Empty
 
-        -- Wait an epoch and make sure the reward doesn't increase
-        epoch <- getFromResponse (#networkTip . #epochNumber) <$>
-            request @ApiNetworkInformation ctx networkInfoEp Default Empty
-        eventually $ do
-            epoch' <- getFromResponse (#networkTip . #epochNumber) <$>
-                request @ApiNetworkInformation ctx networkInfoEp Default Empty
-            unless (getApiT epoch' > getApiT epoch) $ fail "not yet"
+        waitForNextEpoch ctx
         request @ApiWallet ctx (getWalletEp w) Default Empty >>= flip verify
             [ expectFieldSatisfy balanceReward (== reward)
             ]
 
+    it "STAKE_POOLS_JOIN_04 -\
+        \Delegate, stop in the next epoch, and still earn rewards" $ \ctx -> do
+        w <- fixtureWallet ctx
+        (_, p1:_) <- eventually $
+            unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+
+        joinStakePool ctx (p1 ^. #id) (w, fixturePassphrase) >>= flip verify
+            [ expectResponseCode HTTP.status202
+            ]
+
+        waitForNextEpoch ctx
+
+        quitStakePool ctx (p1 ^. #id) (w, fixturePassphrase) >>= flip verify
+            [ expectResponseCode HTTP.status202
+            ]
+
+        eventually $ do
+            request @ApiWallet ctx (getWalletEp w) Default Empty >>= flip verify
+                [ expectFieldSatisfy balanceReward (> 0)
+                ]
+
```

Comment:
Maybe it would be worth also to `waitForNextEpoch ctx` after the last request/assertion and make sure that the rewards are still the same?

Reply:
Agree.

Reply:
See https://github.com/input-output-hk/cardano-wallet/pull/1137/commits/48ca2832d29c21729b76cfaf3b36f4453d3dbfd1